slowfast/models/build.py

```diff
@@ -43,58 +133,50 @@ def build_model(cfg, gpu_id=None):
             cfg.NUM_GPUS == 0
         ), "Cuda is not available. Please set `NUM_GPUS: 0 for running on CPUs."
 
-    # Construct the model
-    name = cfg.MODEL.MODEL_NAME
-    model = MODEL_REGISTRY.get(name)(cfg)
-
-    if cfg.BN.NORM_TYPE == "sync_batchnorm_apex":
-        try:
-            import apex
-        except ImportError:
-            raise ImportError("APEX is required for this model, pelase install")
-
-        logger.info("Converting BN layers to Apex SyncBN")
-        process_group = apex.parallel.create_syncbn_process_group(
-            group_size=cfg.BN.NUM_SYNC_DEVICES
-        )
-        model = apex.parallel.convert_syncbn_model(
-            model, process_group=process_group
-        )
-
+    cur_device = 'cpu'
     if cfg.NUM_GPUS:
         if gpu_id is None:
             # Determine the GPU used by the current process
             cur_device = torch.cuda.current_device()
         else:
             cur_device = gpu_id
-        # Transfer the model to the current GPU device
-        model = model.cuda(device=cur_device)
-    
-    #NOTE: jit Analysis doesn't supprt FSDP wrapped model. Logging is done before wrapping
-    if du.is_master_proc() and cfg.LOG_MODEL_INFO:
-        misc.log_model_info(model, cfg, use_train_input=True)
-
-    # Use multi-process data parallel model in the multi-gpu setting
-    if cfg.NUM_GPUS > 1:
-        if(cfg.FSDP.ENABLED):
-            model = FSDP(
-                model, 
-                reshard_after_forward = cfg.FSDP.RESHARD_AFTER_FW, 
-                mixed_precision=cfg.TRAIN.MIXED_PRECISION,
+
+    if not (cfg.FSDP.ENABLED and cfg.FSDP.NESTED_WRAP):
+        model_name = cfg.MODEL.MODEL_NAME
+        model = MODEL_REGISTRY.get(model_name)(cfg)
+        if cfg.BN.NORM_TYPE == "sync_batchnorm_apex":
```

Comment:
we should crash if a user passes incompatible options like `sync_batchnorm_apex` and FSDP.

Reply:
Handled